### PR TITLE
1529 Refactor how we save channel & force

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🧰 Issue

#1529

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/custom-types/force.d.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export default interface Force {
   // TODO: we need to constrain what we store below.
   // I _think_ we're storing a whole force. I'm sure it should be
   // a force-id
-  selectedForce: string,
+  selectedForce: string | { uniqid : string },
```
Comment (IanMayo) [inline packages/custom-types/force.d.ts]: How about we specify `string | ForceData | undefined`

To be honest, I don't like the `|` union.  How about we remove usage of strings in there, and just use:
```
  selectedForce?: ForceData
```

Aah, but we **certainly don't** want to store the whole `ForceData` object when we persist to database.

File: packages/server/providers/couchdb.js [hunk 2]
```diff
@@ -90,7 +90,15 @@ const couchDb = (app, io, pouchOptions) => {
         if (err.status === 409) {
           return retryUntilWritten(db, doc)
         } else { // new doc
-          return db.put(doc).then(() => res.send({ msg: 'Saved', data: doc }))
```
Comment (IanMayo) [inline packages/server/providers/couchdb.js:93]: Is this fixing a bug @lilitkarapetyan ?

File: packages/server/providers/couchdb.js [hunk 3]
```diff
@@ -161,7 +169,7 @@ const couchDb = (app, io, pouchOptions) => {
     remoteDb.allDocs({ include_docs: true, attachments: true })
       .then(result => {
         const messages = result.rows.reduce((messages, { doc }) => {
-          const isNotSystem = doc._id !== wargameSettings
+          const isNotSystem = doc._id !== wargameSettings || doc._id !== settings
```
Comment (IanMayo) [inline packages/server/providers/couchdb.js]: Hmm, I think the above needs to be `&&`. Both need to be false to be sure it's not the system doc

File: packages/server/providers/couchdb.js (hunk 2 above)
Comment (lilitkarapetyan) [reply packages/server/providers/couchdb.js:93]: Yes, this part is for updating "initial_wargame" and I added "settings" if "initial_wargame" doesn't exist.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**